test/Microsoft.AspNetCore.Routing.Tests/DefaultLinkGeneratorTest.cs

```diff
@@ -239,15 +259,259 @@ public void GetLink_Success_AmbientValues()
+                });
 
             // Assert
             Assert.Equal("/Home/Index", link);
         }
 
+        [Fact]
+        public void GetLink_GeneratesLowercaseUrl_SetOnRouteOptions()
+        {
+            // Arrange
+            var endpoint = CreateEndpoint("{controller}/{action}");
+            var linkGenerator = CreateLinkGenerator(new RouteOptions() { LowercaseUrls = true });
+            var context = CreateRouteValuesContext(
+                suppliedValues: new { action = "Index" },
+                ambientValues: new { controller = "Home" });
+
+            // Act
+            var link = linkGenerator.GetLink(
+                new LinkGeneratorContext
+                {
+                    Endpoints = new[] { endpoint },
+                    ExplicitValues = context.ExplicitValues,
+                    AmbientValues = context.AmbientValues
+                });
+
+            // Assert
+            Assert.Equal("/home/index", link);
+        }
+
+        [Fact]
+        public void GetLink_GeneratesLowercaseQueryString_SetOnRouteOptions()
+        {
+            // Arrange
+            var endpoint = CreateEndpoint("{controller}/{action}");
+            var linkGenerator = CreateLinkGenerator(
+                new RouteOptions() { LowercaseUrls = true, LowercaseQueryStrings = true });
+            var context = CreateRouteValuesContext(
+                suppliedValues: new { action = "Index", ShowStatus = "True", INFO = "DETAILED" },
+                ambientValues: new { controller = "Home" });
+
+            // Act
+            var link = linkGenerator.GetLink(
+                new LinkGeneratorContext
+                {
+                    Endpoints = new[] { endpoint },
+                    ExplicitValues = context.ExplicitValues,
+                    AmbientValues = context.AmbientValues
+                });
+
+            // Assert
+            Assert.Equal("/home/index?showstatus=true&info=detailed", link);
+        }
+
+        [Fact]
+        public void GetLink_GeneratesLowercaseQueryString_OnlyIfLowercaseUrlIsTrue_SetOnRouteOptions()
+        {
+            // Arrange
+            var endpoint = CreateEndpoint("{controller}/{action}");
+            var linkGenerator = CreateLinkGenerator(
+                new RouteOptions() { LowercaseUrls = false, LowercaseQueryStrings = true });
+            var context = CreateRouteValuesContext(
+                suppliedValues: new { action = "Index", ShowStatus = "True", INFO = "DETAILED" },
+                ambientValues: new { controller = "Home" });
+
+            // Act
+            var link = linkGenerator.GetLink(
+                new LinkGeneratorContext
+                {
+                    Endpoints = new[] { endpoint },
+                    ExplicitValues = context.ExplicitValues,
+                    AmbientValues = context.AmbientValues
+                });
+
+            // Assert
+            Assert.Equal("/Home/Index?ShowStatus=True&INFO=DETAILED", link);
```

Comment:
query string is not lowercase?

Reply:
It is lower cased only when LowercaseUrls is also true

Reply:
Is that existing behavior?

Need to make sure that is documented on property

Reply:
Yeah, its an existing behavior (i know it feels weird) https://github.com/aspnet/Routing/blob/release/2.2/src/Microsoft.AspNetCore.Routing/RouteOptions.cs#L20

Reply:
Ok! Copy over those properties xml docs